### PR TITLE
Perm arg hotfix

### PR DESCRIPTION
**Pull request by weeryan17**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] All changes have been tested

## Fixed
An issue where we where trying to get an argument that didn't exist

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java [hunk 1]
```diff
@@ -30,119 +30,121 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
-                        Role role = GeneralUtils.getRole(args[3], guild.getGuildId());
-                        if (role != null) {
-                            group.linkRole(role.getId());
-                            MessageUtils.sendSuccessMessage("Successfully linked the group `" + groupString + "` to the role `" + role.getName() + "`", channel, sender);
-                            return;
-                        } else {
-                            MessageUtils.sendErrorMessage("That role doesn't exist!", channel);
-                            return;
+                    } else if (args[2].equalsIgnoreCase("create")) {
+                        if (args.length == 3) {
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: Not needed, You already check for 3 on the outside

File: src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java [hunk 2]
```diff
@@ -30,119 +30,121 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
-                    if (args.length == 3) {
-                        Role role;
-                        if (group.getRoleId() == null || (role = guild.getGuild().getRoleById(group.getRoleId())) == null) {
-                            MessageUtils.sendErrorMessage("Cannot unlink if a role isn't linked!!", channel);
-                            return;
-                        } else {
-                            group.linkRole(null);
-                            MessageUtils.sendSuccessMessage("Successfully unlinked the role " + role.getName() + " from the group " + group.getName(), channel, sender);
+                    } else if (args[2].equalsIgnoreCase("delete")) {
+                        if (args.length == 3) {
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: Same

File: src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java [hunk 3]
```diff
@@ -30,119 +30,121 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
                             } else {
-                                MessageUtils.sendErrorMessage("That role doesn't exist!!", channel);
+                                MessageUtils.sendErrorMessage("That role doesn't exist!", channel);
                                 return;
                             }
                         }
-                        for (Member user : roleMembers) {
-                            getPermissions(channel).getUser(user).addGroup(group);
+                    } else if (args[2].equalsIgnoreCase("unlink")) {
+                        if (args.length == 3) {
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: Same

File: src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java [hunk 4]
```diff
@@ -30,119 +30,121 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
+                            } else {
+                                group.linkRole(null);
+                                MessageUtils.sendSuccessMessage("Successfully unlinked the role " + role.getName() + " from the group " + group.getName(), channel, sender);
+                                return;
+                            }
                         }
-                        MessageUtils.sendSuccessMessage("Successfully added the group `" + groupString + "` to everyone in the role @" + roleName, channel, sender);
-                        return;
+                    } else if (args[2].equalsIgnoreCase("list")) {
+                        if (args.length == 3 || args.length == 4) {
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: args.length <= 4, you already check for 3 on the outside

File: src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java [hunk 5]
```diff
@@ -30,65 +30,61 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
...
-                            MessageUtils.sendSuccessMessage("Successfully added the permission `" + args[3] + "` to the group `" + groupString + "`", channel, sender);
-                            return;
-                        } else {
-                            MessageUtils.sendErrorMessage("Couldn't add the permission (it probably already exists)", channel);
-                            return;
-                        }
-
+                if (args.length >= 3) {
+                    if (group == null && !args[2].equalsIgnoreCase("create")) {
+                        MessageUtils.sendErrorMessage("That group doesn't exist! You can create it with `" + getPrefix(channel.getGuild()) + "permissions group " + groupString + " create`", channel);
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: Can just do {%}

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: we format those is sendErrorMessage() ?

Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: Nope we don't

Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: Oh yes we do, do what walshy said

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: 🤦‍♂️

**Pull request merged**